### ISSUE TITLE
Cannot read properties of undefined (reading 'externalNodes') on all versions since @nrwl/react-native@14.5.8

### ISSUE THREAD

**Issue by davisantos-tag** [nrwl/nx#11761]
## Current Behavior
Running both **run-ios/run-android** commands is returning an error and not running the react-native apps. 

<img width="530" alt="image" src="https://user-images.githubusercontent.com/106993146/187282429-73ac4bd4-4b7c-496f-afc5-549a1d1fc819.png">
Downgrading @nrwl/react-native package to 14.5.7 seems to be working. But not with the latest versions.

## Expected Behavior
Run the app properly using: `npx nx project:run-ios` or `npx nx project:run-android`

## Steps to Reproduce
1. Update the @nrwl/react-native package to any version from 14.5.8;
2. Try to execute `npx nx project:run-ios` or `npx nx project:run-android`;
3. Get the error.

### Failure Logs
```
TypeError: Cannot read properties of undefined (reading 'externalNodes')
    at findAllNpmDependencies (/Users/davi.santos/Documents/project/node_modules/@nrwl/react-native/src/utils/find-all-npm-dependencies.js:11:24)
    at /Users/davi.santos/Documents/project/node_modules/@nrwl/react-native/src/executors/sync-deps/sync-deps.impl.js:23:78
    at Generator.next (<anonymous>)
    at /Users/davi.santos/Documents/project/node_modules/tslib/tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/davi.santos/Documents/project/node_modules/tslib/tslib.js:114:16)
    at syncDeps (/Users/davi.santos/Documents/project/node_modules/@nrwl/react-native/src/executors/sync-deps/sync-deps.impl.js:22:20)
    at runIosExecutor_1 (/Users/davi.santos/Documents/project/node_modules/@nrwl/react-native/src/executors/run-ios/run-ios.impl.js:20:137)
    at runIosExecutor_1.next (<anonymous>)
    at resume (/Users/davi.santos/Documents/project/node_modules/tslib/tslib.js:230:48)
```

### Environment
<img width="444" alt="image" src="https://user-images.githubusercontent.com/106993146/187282645-e9422c40-ff32-417b-8b64-05f72188a6df.png">

**Comment by AgentEnder** [comment]
Hey @davisantos-tag, it may be unrelated, but some of your @nrwl/* packages don't line up with the version of `nx` you have installed. Aside from `@nrwl/nx-cloud`, all packages under the nrwl scope should generally be set up with the same version. 

Can you correct this and check that the issue still reproduces? The recommended way to fix this would be to run `nx migrate 14.5.10`, to bump to the latest 14.5.X patch version

**Comment by davisantos-tag** [comment]
Hi @AgentEnder, good to know, it worked now. Thank you very much for the quick answer!
